src/coreclr/System.Private.CoreLib/src/System/Reflection/MethodInvoker.CoreCLR.cs

```diff
@@ -15,13 +15,16 @@ public MethodInvoker(MethodBase method, Signature signature)
             _method = method;
             _signature = signature;
 
-#if USE_NATIVE_INVOKE
-            // Always use the native invoke; useful for testing.
-            _strategyDetermined = true;
-#elif USE_EMIT_INVOKE
-            // Always use emit invoke (if IsDynamicCodeCompiled == true); useful for testing.
-            _invoked = true;
-#endif
+            if (LocalAppContextSwitches.ForceInterpretedInvoke && !LocalAppContextSwitches.ForceEmitInvoke)
+            {
+                // Always use the native invoke; useful for testing.
+                _strategyDetermined = true;
```

Comment:
Just for my understanding: it seems the `interpreted(native) invoke` is the old (default) invoke, I wonder why we would need a switch for testing this, and add a new test project for this,  isn't the existing tests already testing the old behavior without any extra handling? I mean it seems to me only one switch for new `emit invoke` could be enough

Reply:
Yes the existing tests will verify the native\interpreted, but only if the particular method has not been invoked before in the same suite, so it is somewhat flaky to depend on this especially if we ever change the algorithm going forward,